File: solutions/Question 3005/Solution2.py
# Problem: 3005. Count Elements With Maximum Frequency
# Difficulty: Easy
# Topics: Array, Hash Table, Counting
# URL: https://leetcode.com/problems/count-elements-with-maximum-frequency/
# Runtime: 3 ms
# Memory: 17.8 MB
# Solved via CodePath Auto-Committer

from collections import Counter
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        freq = Counter(nums)  # frequency of each element
        max_freq = max(freq.values())  # maximum frequency
        count = sum(v for v in freq.values() if v == max_freq)  # sum of elements with max freq
        return count

File: solutions/Question 1480/Solution2.cpp
/*
 * Problem: 1480. Running Sum of 1d Array
 * Difficulty: Easy
 * Topics: Array, Prefix Sum
 * URL: https://leetcode.com/problems/running-sum-of-1d-array/
 * Runtime: 0 ms
 * Memory: 12.5 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> ans;
        int num = 0;
        for(int x: nums){
            num = num + x;
            ans.push_back(num);
        }
        return ans;
    }
};